prov/shm/src/smr_rma.c

```diff
@@ -191,8 +204,6 @@ static ssize_t smr_generic_rma(struct smr_ep *ep, const struct iovec *iov,
 	smr_signal(peer_smr);
 unlock_cq:
 	ofi_spin_unlock(&ep->tx_lock);
```

Comment:
Through logging, it appears this unlocking op was deadlocked in this test
```
efa/test_rma_bw.py::test_rma_bw[transmit_complete-host_to_host-standard-write]
```
Any idea?

Reply:
I think I see the issue:
```
	ret = smr_cmd_queue_next(smr_cmd_queue(peer_smr), &ce, &pos);
	if (ret == -FI_ENOENT) {
		/* kick the peer to process any outstanding commands */
		smr_signal(peer_smr);
		return -FI_EAGAIN;
	}
```
Function returns here without unlocking the tx_lock.
Will do another pass over the code to ensure we don't have a similar issue somewhere else.
thanks for the pointer.

Reply:
Further clarification - what I observed is that `ofi_spin_unlock(&ep->tx_lock);` was run but did not return, in case I wasn't clear.

Reply:
Weird. missing an unlock should cause the next lock operation to block, not the unlock.

Reply:
I'm puzzled by that as well. Not sure how unlock would get stuck, unless the value of the lock is corrupted. We'll see if we still get the same issue with the fix I posted.

Reply:
Yes. It also smells weird to me - maybe the logging is wrong? Let's see if it works after this patch.